src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,89 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int GROW_FACTOR = 2;
 
-    @Override
-    public void put(K key, V value) {
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (table.length * LOAD_FACTOR);
+    }
 
+    public void put(K key, V value) {
+        int index = getIndexForKey(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (size >= threshold) {
+            resize();
+        }
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> node = table[index];
+        while (true) {
+            if (Objects.equals(key, node.key)) {
+                node.value = value;
+                return;
+            }
+            if (node.next == null) {
+                node.next = newNode;
+                size++;
+                return;
+            }
+            node = node.next;
+        }
     }
 
-    @Override
     public V getValue(K key) {
+        int index = getIndexForKey(key);
+        Node<K, V> node = table[index];
+        while (node != null) {
+            if (Objects.equals(key, node.key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
-    @Override
```

Comment:
do not remove the `Override` annotations, they may not be necessary but bring more readability